control-plane-operator/controllers/hostedcontrolplane/kas/aws_kms.go

```diff
@@ -28,12 +30,18 @@ var (
 			kasVolumeKMSSocket().Name: "/var/run",
 		},
 		kasContainerAWSKMSActive().Name: {
-			kasVolumeKMSSocket().Name:         "/var/run",
-			kasVolumeAWSKMSCredentials().Name: "/.aws",
+			kasVolumeKMSSocket().Name:                "/var/run",
+			kasVolumeAWSKMSCredentials().Name:        "/aws",
+			kasVolumeAWSKMSCloudProviderToken().Name: "/var/run/secrets/openshift/serviceaccount",
 		},
 		kasContainerAWSKMSBackup().Name: {
-			kasVolumeKMSSocket().Name:         "/var/run",
-			kasVolumeAWSKMSCredentials().Name: "/.aws",
+			kasVolumeKMSSocket().Name:                "/var/run",
```

Comment:
Just for my understanding, why is there an active and a backup container for this? And their definitions seem to be equal, is there benefit to duplicating it?

Reply:
The backup key is to enable key rotation. That way you can at least transition by decrypting content with the old key and encrypting with the new key. Each KMS provider implements this in their own way. Some providers allow you to specify and active and backup key together. However, the AWS KMS provider only supports a single key per container. So if you want both keys available, you need separate containers and the configuration pointing to each of their endpoints.

Reply:
So the encryption will always use the active container and the decryption will default to the active container and fallback to the backup one in the AWS case?

Reply:
yes